sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/_polling.py

```diff
@@ -0,0 +1,166 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import logging
+import time
+import threading
+import uuid
+from typing import TYPE_CHECKING
+
+from azure.core.polling import PollingMethod, LROPoller
+from azure.core.exceptions import ResourceNotFoundError, HttpResponseError
+
+try:
+    from urlparse import urlparse # type: ignore # pylint: disable=unused-import
+except ImportError:
+    from urllib.parse import urlparse
+
+from azure.core.pipeline.transport._base import HttpResponse  # type: ignore
+from azure.core.tracing.decorator import distributed_trace
+from azure.core.tracing.common import with_current_context
+
+if TYPE_CHECKING:
+    # pylint: disable=ungrouped-imports
+    import requests
+    from msrest.serialization import Model # type: ignore # pylint: disable=unused-import
+    from typing import Any, Callable, Union, List, Optional
+    DeserializationCallbackType = Union[Model, Callable[[requests.Response], Model]]
+
+logger = logging.getLogger(__name__)
+
+
+class KeyVaultOperationPoller(LROPoller):
+    """Poller for long running operations where calling result() doesn't wait for operation to complete.
+
+    :param client: A pipeline service client
+    :type client: ~azure.core.PipelineClient
+    :param initial_response: The initial call response
+    :type initial_response:
+     ~azure.core.pipeline.transport.HttpResponse or ~azure.core.pipeline.transport.AsyncHttpResponse
+    :param deserialization_callback: A callback that takes a Response and return a deserialized object.
+                                     If a subclass of Model is given, this passes "deserialize" as callback.
+    :type deserialization_callback: callable or msrest.serialization.Model
+    :param polling_method: The polling strategy to adopt
+    :type polling_method: ~azure.core.polling.PollingMethod
+    """
+
+    def __init__(self, client, initial_response, deserialization_callback, polling_method):
+        # type: (Any, HttpResponse, DeserializationCallbackType, PollingMethod) -> None
+        # pylint: disable=super-init-not-called
+        self._client = client
+        self._response = initial_response
```

Comment:
If I understand this correctly, `deleted_resource` is probably a better name than `initial_response`, and it's only needed to initialize the polling method, so needn't be assigned to an attribute here?

Reply:
moved from passing final resource to poller to passing final resource to polling method, thanks!